Add hide_for method to Comment class

pfefferle opened the pull request:
## Proposed changes:

* Add new `hide_for()` method to the `Comment` class that returns post types whose comments should be hidden in the WordPress admin.
* Refactor `comment_query()` and `pre_comment_approved()` methods to use the new helper method.
* Add `activitypub_hide_comments_for` filter to customize which post types have hidden comments.
* Add comprehensive unit tests for the new functionality and filter behavior.

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Run the test suite: `npm run env-test -- --filter=test_hide_for`
* Verify the tests pass and confirm:
  * `ap_post` is the only post type in the default list
  * The filter can add/remove post types
  * The filter affects both comment visibility and auto-approval

### Changelog entry

Changelog entry added manually via `composer changelog:add`.

pfefferle committed:
Add get_post_types_to_hide_comments_for method to Comment class.

This method returns post types whose comments should be hidden in the
WordPress admin. Used for filtering comment queries and auto-approving
comments on non-public post types like ap_post.

jeherve commented on includes/class-comment.php:
```diff
@@ -704,9 +703,7 @@ public static function comment_query( $query ) {
-			if ( \get_option( 'activitypub_create_posts', false ) ) {
-				$query->query_vars['post_type'] = array_diff( \get_post_types_by_support( 'comments' ), array( Posts::POST_TYPE ) );
-			}
+			$query->query_vars['post_type'] = array_diff( \get_post_types_by_support( 'comments' ), self::get_post_types_to_hide_comments_for() );
```
Would it be an option to limit that to `ap_post`? What other post types do we want to grab with this change?

pfefferle replied:
If you can verify that this is "only" a problem with `ap_post` and there are no further interference with the friends plugin, sure!

pfefferle committed:
Add hide_for method to Comment class.

This method returns post types whose comments should be hidden in the
WordPress admin comments list.

jeherve reviewed (dismissed):
This looks good and tests well for me. I like the addition of the filter, it should allow third-parties to hook into that logic if necessary.

pfefferle committed:
Merge branch 'trunk' into add/get-non-public-post-types
